### ISSUE TITLE
FIDO Resident Keys

### ISSUE THREAD

**Issue by schichtnudelauflauf** [mooltipass/minible#353]
### Missing feature

When trying to provision a resident key the device says its unsupported.

### Justification
My usecase is ssh-keygen's ed25519-sk keys. Because that works with normal ssh-agent.
I do not want to use any custom agent because that precludes usage of other keybearer devices.
I do not want to use the non resident fido keys either because that requires moving auxiliary files around.

### Workarounds
None

**Comment by My1** [comment]
I think the issue is not resident keys as as far as I can see ALL FIDO2 keys on MP are resident (something which I would love not to be the case), but rather the fact that the MP uses internal UV, without fallback to clientPIN (as in, a PIN you literally enter on PC).

basically similar to biometric keys which can use fingerprints in lieu of a clientPIN, but equally fall back to that clientpin if either
1) an application cannot deal with internal UV
2) your fingerprint was not recognized a few times
the MP only uses internal UV, basically allowing its unlocked state to count as user verification, but cannot fallback to a clientPIN.

the FIDO client within openSSH does not seem to deal with internal UV yet, at least the one I have here.

**Comment by My1** [comment]
Update: after running it with verbose it is clear what's going on:
```
my1@my1-qbp:~$ ssh-keygen -v -t ed25519-sk -O resident -f /tmp/key
Generating public/private ed25519-sk key pair.
You may need to touch your authenticator to authorize key generation.
Enter PIN for authenticator: 
debug1: start_helper: starting /usr/lib/openssh/ssh-sk-helper 
debug1: sshsk_enroll: provider "internal", device "(null)", application "ssh:", userid "(null)", flags 0x21, challenge len 0
debug1: sshsk_enroll: using random challenge
debug1: sk_probe: 1 device(s) detected
debug1: sk_probe: selecting sk by touch
debug1: ssh_sk_enroll: using device /dev/hidraw3
debug1: ssh_sk_enroll: /dev/hidraw3 does not support credprot, refusing to create unprotected resident/verify-required key
debug1: sshsk_enroll: provider "internal" failure -2
debug1: ssh-sk-helper: Enrollment failed: requested feature not supported
debug1: main: reply len 8
debug1: client_converse: helper returned error -59
Key enrollment failed: requested feature not supported
```

TLDR the credProtect feature is apparently something openSSH asks for which the MP does not have set as active yet (although considering the way the MP handles UV it's always active

**Comment by 0x0ptr** [comment]
Yes, @My1 is correct. We do not support CredProtect at this time. As far as I remember that takes some considerable work to enable (several dependencies).

**Comment by My1** [comment]
Yeah i guess. I mean implicitly this device already has credential protection (as it's literally impossible to read anything without it being unlocked) but credprotect is iirc part of ctap2.1 for the first time and ctap2.1 brings along several requirements, some of which (iirc hmac says hello) are also issues here

**Comment by spoelstraethan** [comment]
I would love to see the Mooltipass BLE support resident keys fully (the more open source projects that can do so the better).

I ran across this AMAZING description of how to get these working with OpenSSH 8.3+ and it might be helpful for other scenarios as well.

https://github.com/betrusted-io/xous-core/issues/209#issuecomment-1203338599

**Comment by My1** [comment]
as mentioned resident keys in itself fully works. credprotect is a much newer feature which came on CTAP2.1 (and also devices that went ahead and implemented the pre version), while the MP is basically running CTAP2.0 with ALL keys being resident in fact.

this is the same as if you took an older yubikey 5 which was before all the ctap2.1 stuff, it certainly has resident keys but would be too old to have credprotect.

credprotect is not a part of resident keys, it's its own thing.
